planetary-variables/soil-water-content/soil-water-content-flash-drought/raw.js

```diff
@@ -0,0 +1,82 @@
+//VERSION=3
+
+const nDaysBackwardAverage = 14; // The number of days that is used to calculate the swc backward average
+const nDaysPrevious = 28; // The number of days to look back to compare the current value with the previous value
+const scaleFactor = 1000; // The scale factor for the SWC values
+const droughtThreshold = 0.15; // The SWC value under which conditions are considered drought-like
+const differenceThreshold = -0.12; // Threshold signifying a rapid drop in SWC from the previous average value
+
+function setup() {
+  return {
+    input: ["SWC", "dataMask"],
+    output: { bands: 1 },
+    mosaicking: "ORBIT",
+  };
+}
+
+const daysToLoad = nDaysBackwardAverage + nDaysPrevious; // The number of days looking back to load data for
+
+function preProcessScenes(collections) {
+  const millisecondsBack = daysToLoad * 24 * 3600 * 1000;
+
+  collections.scenes.orbits = collections.scenes.orbits.filter(function (
+    orbit
+  ) {
+    const orbitDateFrom = new Date(orbit.dateFrom);
+    return (
+      orbitDateFrom.getTime() >= collections.to.getTime() - millisecondsBack
+    );
+  });
+  return collections;
+}
+
+function getMeanSWCValue(swc, dataMask) {
+  // Get the sum of all SWC values
+  let validDateCount = 0;
+  let sum = 0;
+  for (let i = 0; i < swc.length; i++) {
+    if (dataMask[i]) {
+      sum += swc[i];
```

Comment:
To make it work without extracting the values from samples a step earlier:

```suggestion
function getMeanSWCValue(samples) {
  // Get the sum of all SWC values
  let validDateCount = 0;
  let sum = 0;
  for (let i = 0; i < samples.length; i++) {
    if (samples[i].dataMask) {
      sum += samples[i].swc;
```